repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -226,7 +226,13 @@ class Interpreter(prompt0: Ref[String],
         eval.addJar(jar.toURI.toURL)
       }
 
-      def apply(line: String) = processExec(line)
+      def apply(line: String, visible: Boolean) = {
+        if (!visible) processExec(line)
```

Comment:
OK... I tried having `processExec` take an optional `stdout` argument, and I could not figure out a straightforward way to make it work:

*  `processExec` just calls `processScript`, `processScript` itself does not take any kind of printer.
*  `processExec` does pass `processScript` an `evaluateLine` function, which _does_ take a printer:  `{ (c, i) => evaluateLine(c, Nil, _ => () , i)}` But when I replace the `_ => ()` argument with an actual printer, I still don't get anything printed out.
* the signature for `evaluateLine` is : 

```
def evaluateLine(code: String,
                   printSnippet: Seq[String],
                   printer: Iterator[String] => Unit,
                   extraImports: Seq[ImportData] = Seq() ) 
```

*  I suspect I need to replace the `Nil` value passed to `evaluateLine` for the `printSnippet` with something to get printing to work, but I'm not sure what! 
* Here is what I have so far:

```
def apply(line: String, visible: Boolean) = {
     if (!visible) processExec(line) else processExec(line, _.foreach(interp.stdout)) 
     /*
     else {
        val res = interp.processLine(line, Parsers.split(line).get.get.value, _.foreach(interp.stdout))
        handleOutput(res)
     }
     */
}

def processExec(code: String, printer: Iterator[String] => Unit = _ => () ) =
    processScript(prepareScript(code), { (c, i) => evaluateLine(c, Nil, printer, i)})
```

Any suggestions to to where to go from here?